opnode/node/driver_test.go

```diff
@@ -0,0 +1,157 @@
+package node
+
+import (
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+// ================================================================================================
+// INPUT GENERATION HELPERS
+
+var (
+	zero        = big.NewInt(0)
+	zeroHash    = common.BigToHash(zero)
+	zeroAddress = common.BigToAddress(zero)
+	chaindID    = big.NewInt(69) // testnet chain ID, not meaningful here
+)
+
+var depositEventABI = "TransactionDeposited(address,address,uint256,uint256,bool,bytes)"
+
+func GenerateInputItem() InputItem {
+	txs := types.Transactions{GenerateTransaction()}
+	receipts := types.Receipts{}
+	block := GenerateBlock(txs, receipts, time.Now())
+	return InputItem{
+		Block:    block,
+		Receipts: receipts,
+	}
+}
+
+func GenerateTransaction() *types.Transaction {
+	txData := &types.DynamicFeeTx{
+		ChainID: chaindID,
+		Data:    []byte{},
+
+		// ignored (zeroed):
+		Nonce:      0,
+		GasTipCap:  zero,
+		GasFeeCap:  zero,
+		Gas:        0,
+		To:         &zeroAddress,
+		Value:      zero,
+		AccessList: types.AccessList{},
+		V:          zero,
+		R:          zero,
+		S:          zero,
+	}
+	return types.NewTx(txData)
+}
+
+func GenerateDepositLog() *types.Log {
+	// as specified by https://docs.soliditylang.org/en/v0.8.10/abi-spec.html?highlight=topics#events
+
+	topics := []common.Hash{
+		common.BytesToHash(crypto.Keccak256([]byte(depositEventABI))),
+	}
+	data := []byte{}
+	return GenerateLog(DepositContractAddr, topics, data)
+}
+
+func GenerateLog(addr common.Address, topics []common.Hash, data []byte) *types.Log {
+	return &types.Log{
+		Address: addr,
+		Topics:  topics,
+		Data:    data,
+		Removed: false,
+
+		// ignored (zeroed):
+		BlockNumber: 0,
+		TxHash:      zeroHash,
+		TxIndex:     0,
+		BlockHash:   zeroHash,
+		Index:       0,
+	}
+}
+
+func GenerateReceipt() *types.Receipt {
+	logs := []*types.Log{}
+	receipt := &types.Receipt{
+		Type:   types.DynamicFeeTxType,
+		Status: types.ReceiptStatusSuccessful,
+		Logs:   logs,
+
+		// ignored (zeroed):
+		PostState:         []byte{},
+		CumulativeGasUsed: 0,
+		TxHash:            zeroHash,
+		ContractAddress:   zeroAddress,
+		GasUsed:           0,
+		BlockHash:         zeroHash,
+		BlockNumber:       zero,
+		TransactionIndex:  0,
```

Comment:
If fields are zeroed, you can just omit them completely. (the byte slice will be `nil`, but generally acts the same, and allocates a slice when necessary on first append). No need to specify a `zeroHash` or `zeroAddress`! (And if you really need to be explicit, `common.Address{}` and `common.Hash{}` would be the idiomatic Go way.

Reply:
Yup, it was my intention to be explicit, in case we needed to tweak some other fields in the generation process.

Good stuff about `common.Address{}`, only realized you could do that with array aliases later!